### PR TITLE
DOCSP-43088: Aggregations

### PR DESCRIPTION
**Pull request by mcmorisi**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-golang/blob/master/REVIEWING.md)

JIRA - <https://jira.mongodb.org/browse/DOCSP-43088>
Staging - <https://deploy-preview-38--docs-c.netlify.app/aggregation/>

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?
- [x] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/aggregation.txt [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+:atlas:`Get Started with Atlas </getting-started>` guide.
+
+Build and Execute an Aggregation Pipeline
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To perform an aggregation on the documents in a collection, pass a ``bson_t`` structure
+that represents the pipeline stages to the ``mongoc_collection_aggregate()`` function.
+
+This example outputs a count of the number of bakeries in each borough
+of New York City. The following code creates aggregation pipeline that contains the
```
Comment (jordan-smith721) [inline source/aggregation.txt:92]:
```suggestion
of New York City. The following code creates an aggregation pipeline that contains the
```

**Review by jordan-smith721 — APPROVED** [review]
LGMT w/ one fix

File: source/includes/aggregation/aggregation.c [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+            "]");
+
+        mongoc_cursor_t *results =
+            mongoc_collection_aggregate (collection, MONGOC_QUERY_NONE, pipeline, NULL, NULL);
+        
+        while (mongoc_cursor_next (results, &doc)) {
+            char *str = bson_as_canonical_extended_json (doc, NULL);
+            printf ("%s\n", str);
+            bson_free (str);
+        }
```
Comment (kevinAlbs) [inline source/includes/aggregation/aggregation.c:35]: If error handling is relevant to the example (as it is below) suggest adding:
```c
bson_error_t error;
if (mongoc_cursor_error(results, &error))
{
    fprintf(stderr, "Aggregate failed: %s\n", error.message);
}
```

**Review by kevinAlbs — APPROVED** [review]
LGTM with possible error check.

**Pull request merged**